lix/packages/fs/src/utilities/helpers.test.ts

```diff
@@ -0,0 +1,100 @@
+import { describe, it, expect, test } from "vitest"
+import { normalizePath, getBasename, getDirname } from "./helpers.js"
+
+describe("single slash", () => {
+	it("should always return a single forward slash", () => {
+		expect(normalizePath("/")).toBe("/")
+		expect(normalizePath("\\")).toBe("/")
+	})
+})
+
+test("should not fail on normalizing the root with an absolute path", () => {
+	expect(normalizePath("/" + "/project.inlang")).toBe("/project.inlang")
+})
+
+test("not add a slash", () => {
+	expect(normalizePath("project.inlang")).toBe("project.inlang")
+})
+
+test("should add leading slash", () => {
+	expect(normalizePath("project.inlang", { leadingSlash: 'always'})).toBe("/project.inlang")
+})
+
+test("should add trailing slash", () => {
+	expect(normalizePath("project.inlang", { trailingSlash: 'always' })).toBe("project.inlang/")
+})
+
+describe("strip trailing slashes", () => {
+	const units: [string, string][] = [
+		["../../foo/bar", "/foo/bar"],
+		["..\\..\\foo/bar", "/foo/bar"],
+		["..\\\\..\\\\foo/bar", "/foo/bar"],
+		["//foo/bar\\baz", "/foo/bar/baz"],
+ 		["//foo\\bar\\baz", "/foo/bar/baz"],
+		["/user/docs/Letter.txt", "/user/docs/Letter.txt"],
+		["\\\\.\\CdRomX", "/CdRomX"],
+		["\\\\.\\PhysicalDiskX", "/PhysicalDiskX"],
+		["\\Server01\\user\\docs\\Letter.txt", "/Server01/user/docs/Letter.txt"],
+		["foo\\bar\\baz", "/foo/bar/baz"],
+		["foo\\bar\\baz\\", "/foo/bar/baz/"],
+		["foo\\bar\\baz\\\\\\", "/foo/bar/baz/"],
+	]
+
+	for (const unit of units) {
+		it(`should normalize ${unit[0]}`, () => {
+			expect(normalizePath(unit[0], { leadingSlash: 'always' })).toBe(unit[1])
+		})
+	}
+})
+
+describe("keep trailing slashes", () => {
+	const units: [string, string][] = [
+		["\\", "/"],
+ 		["foo\\bar\\baz\\", "/foo/bar/baz/"],
+	 	["foo\\\\bar\\\\baz\\\\", "/foo/bar/baz/"],
+		["foo//bar//baz//", "/foo/bar/baz/"],
+		["foo/bar/baz/", "/foo/bar/baz/"],
+		["./foo/bar/baz/", "/foo/bar/baz/"],
```

Comment:
Can we aslso get an example with `./` in the middle of a path? I don't see this tested here, only `../`.